org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -555,4 +582,26 @@ public static InnovationContext getInnovationContext(CodeActionParams params, IP
 	public static ICompilationUnit getCompilationUnit(CodeActionParams params) {
 		return JDTUtils.resolveCompilationUnit(params.getTextDocument().getUri());
 	}
+
+	public static Optional<Either<Command, CodeAction>> copySourceActionCommand(Optional<Either<Command, CodeAction>> sourceActionCommand) {

```

Comment:
Since this utility is not for source action only, i would use a more generic method name such as
```java
public static Optional<Either<Command, CodeAction>> copyCodeActionCommand(Optional<Either<Command, CodeAction>> codeActionCommand)
```

Reply:
Sure, I'll update the method name.

Reply:
According to https://github.com/eclipse/eclipse.jdt.ls/pull/1842#discussion_r692679964, I'd like to remove this method since currently we don't need it.